src/pages/workspaces/workspace/ShareWorkspaceModal/Collaborator.ts

```diff
@@ -0,0 +1,156 @@
+import _ from 'lodash/fp';
+import React, { Dispatch, SetStateAction } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
+import { LabeledCheckbox, Link, Select, SelectProps } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { getPopupRoot } from 'src/components/popup-utils';
+import colors from 'src/libs/colors';
+import { getUser } from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+import { BaseWorkspace, isAzureWorkspace, WorkspaceAccessLevel } from 'src/libs/workspace-utils';
+import { AccessEntry, WorkspaceAcl } from 'src/pages/workspaces/workspace/WorkspaceAcl';
+
+interface CollaboratorProps {
+  aclItem: AccessEntry; // the item to render
+  acl: WorkspaceAcl; // the entire current acl
+  setAcl: Dispatch<SetStateAction<WorkspaceAcl>>;
+  originalAcl: WorkspaceAcl; // the original acl, to determine new items
+  workspace: BaseWorkspace; // the workspace the acl belongs to
+  lastAddedEmail?: string; // the most recently added email to the list
```

Comment:
Curious why this is optional but [this](https://github.com/DataBiosphere/terra-ui/pull/3987/files#diff-1a0932f630136f9c095c8da485b6d3c4e319a37ec82970238f74ab954b6cba5bR11) similarly named field is `string|undefined`